### PR TITLE
Added check for result correctness for Math functions

### PR DESCRIPTION
**Pull request by kopylovanat**
Remaining gap (test coverage): result correctness isn't checked for Integer functions, Geometric Functions

### CODE REVIEW

File: tests/math_builtin_api/math_builtin.h [hunk 1]
```diff
@@ -21,10 +23,25 @@ inline cl::sycl::queue makeQueueOnce() {
...
-template <> inline cl::sycl::half eps<cl::sycl::half>() {
-  return static_cast<cl::sycl::half>(FLT_EPSILON * powf(2.0f, 13.0f));
+template <typename T> struct base;
+template <> struct base<float> { using type = std::uint32_t; };
+template <> struct base<double> { using type = std::uint64_t; };
+template <> struct base<cl::sycl::half> { using type = std::uint16_t; };
+
+template <typename T> std::string printable(T value) {
+  const auto representation =
+      *reinterpret_cast<typename base<T>::type *>(&value);
```
[the remark this answers is not in the document]
Comment (kopylovanat) [reply tests/math_builtin_api/math_builtin.h]: Changed to `sycl::bit_cast`, thanks!

**Review by keryell — APPROVED** [review]
Thanks.

**Review by ProGTX — APPROVED** [review]

**Comment by bader** [comment]
@kopylovanat, could you resolve merge conflicts, please?

**Comment by kopylovanat** [comment]
> @kopylovanat, could you resolve merge conflicts, please?

@bader Done.

**Pull request merged**